# Cutting Planes Example

max $5.5x_1$+$2.1x_2$ <br>
    subject to <br>
    $-x_1+x_2 \leq 2$ <br>
    $8x_1+2x_2 \leq 19$ <br>
    $x_1,x_2 \geq 0$ <br>
    $x_1,x_2$ integer

Load in functions from PuLP package

In [1]:
from pulp import *

We create the model as follows:

In [2]:
model = LpProblem(name="CuttingPlanes", sense=LpMaximize)

We initialise the decision variables

In [3]:
x = {i: LpVariable(name=f"x{i}", lowBound=0, cat="Integer") for i in range(1, 3)}

We add in the objective function

In [4]:
model += 5.5 * x[1] + 2.1 * x[2]

...and the constraints

In [5]:
model += -1 * x[1] + x[2] <= 2
model += 8 * x[1] + 2 * x[2] <= 19

Now we can find the solution of the problem

In [6]:
status = model.solve()

In [8]:
print(f"status: {LpStatus[model.status]}")
print(f"objective: {model.objective.value()}")
for var in x.values():
    print(f"{var.name}: {var.value()}")

status: Optimal
objective: 13.1
x1: 2.0
x2: 1.0


So the optimal solution is $x_1=2,x_2=1$ with an objective value of 13.1